sql/sql_connect.cc

```diff
@@ -1000,6 +1000,28 @@ static int check_connection(THD *thd)
     bzero((char*) &net->vio->remote, sizeof(net->vio->remote));
   }
   vio_keepalive(net->vio, TRUE);
+
+  if (global_system_variables.keepalive_probes != 9)
```

Comment:
Maybe we can live without keepalive probes at all. It is the least portable option (keepalive timeout is the most portable, can be used in Windows and OSX). BTW,  you're aware that in this form the patch is Linux-only and won't even compile elsewhere, right=